crates/oxc_linter/src/rules/eslint/no_negated_condition.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{context::LintContext, rule::Rule, AstNode};
+use oxc_ast::ast::{ConditionalExpression, Expression, IfStatement, Statement};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+use oxc_syntax::operator::{BinaryOperator, UnaryOperator};
+
+fn no_negated_condition_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Unexpected negated condition.").with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoNegatedCondition;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// This rule disallows the use of negated conditions in `if` statements to improve readability.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Negated conditions can make code harder to read and understand, especially in complex logic.
+    /// It is often clearer to use positive conditions or to refactor the code structure.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// if (!isReady) {
+    ///   doSomething();
+    /// } else {
+    ///   doSomethingElse();
+    /// }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// if (isReady) {
+    ///   doSomethingElse();
+    /// } else {
+    ///   doSomething();
+    /// }
+    /// ```
+    NoNegatedCondition,
+    style,
+);
+
+impl Rule for NoNegatedCondition {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::IfStatement(if_stmt) => {
+                if !has_else_without_condition(if_stmt) {
+                    return;
+                }
+
+                if is_negated_if(if_stmt) {
+                    ctx.diagnostic(no_negated_condition_diagnostic(node.span()));
+                }
+            }
+            AstKind::ConditionalExpression(conditional_expr) => {
+                if is_negated_if_conditional(conditional_expr) {
+                    ctx.diagnostic(no_negated_condition_diagnostic(node.span()));
```

Comment:
```suggestion
                    ctx.diagnostic(no_negated_condition_diagnostic(conditional_expr.span));
```

Reply:
Updated!: https://github.com/oxc-project/oxc/pull/8149/commits/145014607b69f6e49c7c40a7526595d545683276

Reply:
@baseballyama you do not need to call `.span()`. `.span` is enough.
Then you can remove the import `use oxc_span::GetSpan;`. 
Sorry I should have told you what my indentations are.

Reply:
Ohhhh; I didn't see this carefully🙏
Updated: https://github.com/oxc-project/oxc/pull/8149/commits/155d9ea0fc90f90dc2709b36ceaf4b5af8b044a3

And maybe there are same issue on other PRs, so I will check this.